packages/expo/src/errors/AppEntryNotFound.tsx

```diff
@@ -0,0 +1,34 @@
+import { StyleSheet, Text, View } from 'react-native';
+
+export function AppEntryNotFound() {
+  return (
+    <View style={styles.container}>
+      <Text style={styles.errorTitle}>App entry not found</Text>
+      <Text style={styles.errorDescription}>
+        The app entry point "main" was not registered. It might be caused by an uncaught error
+        thrown from the top-level code. See the logs from CLI for more details.
+      </Text>
```

Comment:
```suggestion
        The app entry point named "main" was not registered. This may be due to an uncaught error
        thrown from a module's top-level code. Refer to the CLI logs and the native device logs for more detail.
      </Text>
```

(incorporated some of Brent's suggestions too)